inference/core/workflows/core_steps/sinks/modbus_tcp/v1.py

```diff
@@ -0,0 +1,227 @@
+    BATCH_OF_INSTANCE_SEGMENTATION_PREDICTION_KIND,
+    BATCH_OF_KEYPOINT_DETECTION_PREDICTION_KIND,
+    BATCH_OF_OBJECT_DETECTION_PREDICTION_KIND,
+    BATCH_OF_STRING_KIND,
+    BOOLEAN_KIND,
+    ROBOFLOW_PROJECT_KIND,
+    STRING_KIND,
+    ImageInputField,
+    StepOutputImageSelector,
+    StepOutputSelector,
+    WorkflowImageSelector,
+    WorkflowParameterSelector,
+)
+from inference.core.workflows.prototypes.block import (
+    BlockResult,
+    WorkflowBlock,
+    WorkflowBlockManifest,
+)
+
+TYPE: str = "roboflow_core/modbus_tcp@v1"
+SHORT_DESCRIPTION = "Read and Write to a PLC using Modbus TCP"
+LONG_DESCRIPTION = """
+Block to read and write to a PLC using Modbus TCP. This block can be used to read and write to
+holding registers in a PLC. The block can be configured to read or write to the PLC based off 
+predictions from a model.
+"""
+
+
+class ModbusTCPManifest(WorkflowBlockManifest):
+    type: Literal[f"{TYPE}", "ModbusTCP"]
+    model_config = ConfigDict(
+        json_schema_extra={
+            "name": "Modbus TCP",
+            "version": "v1",
+            "short_description": SHORT_DESCRIPTION,
+            "long_description": LONG_DESCRIPTION,
+            "license": "Apache-2.0",
+            "block_type": "communication",
+        }
+    )
+
+    predictions: StepOutputSelector(
+        kind=[
+            BATCH_OF_OBJECT_DETECTION_PREDICTION_KIND,
+            BATCH_OF_INSTANCE_SEGMENTATION_PREDICTION_KIND,
+            BATCH_OF_KEYPOINT_DETECTION_PREDICTION_KIND,
+        ]
+    ) = Field(  # type: ignore
+        description="Predictions",
+        examples=["$steps.instance_segmentation_model.predictions"],
+    )
+
+    comm_type: Union[
+        WorkflowParameterSelector(kind=[STRING_KIND]), Literal["Read", "Write"]
+    ] = Field(
+        description="Read or Write to PLC", examples=["Read", "$inputs.comm_type"]
+    )
+
+    plc_ip: Union[WorkflowParameterSelector(kind=[STRING_KIND]), str] = Field(
+        description="IP Address for the PLC. Must be in format 'xxx.xxx.xxx.xxx'.",
+        examples=["192.168.0.12", "$inputs.plc_ip"],
+    )
+
+    plc_port: Union[WorkflowParameterSelector(kind=[STRING_KIND]), str] = Field(
+        description="Port for the PLC. Must be in format 'xxxx'.",
+        default=502,
+        examples=["502", "$inputs.plc_port"],
+    )
+
+    @classmethod
+    def get_execution_engine_compatibility(cls) -> Optional[str]:
+        return ">=1.0.0,<2.0.0"
+
+    @classmethod
+    def describe_outputs(cls) -> List[OutputDefinition]:
+        return [
+            OutputDefinition(name="status", kind=[STRING_KIND]),
+        ]
+
+    # @classmethod
+    # def describe_outputs(cls) -> List[OutputDefinition]:
+    #     return [
+    #         OutputDefinition(name="error_status", kind=[BATCH_OF_BOOLEAN_KIND]),
+    #         OutputDefinition(name="message", kind=[BATCH_OF_STRING_KIND]),
+    #     ]
+
+
+class ModbusTCPBlockV1(WorkflowBlock):
```

Comment:
if the read and write mode is kind of "simplex" - probably 2 blocks should be created

Reply:
Removed this block.